### PR TITLE
ATL-1118: Add keymaps customization support

### PR DESCRIPTION
**Pull request by fstasi**
The default keybindings may be reasonable for beginners, but PRO users expect the keybindings to be customizable, as they are in other editors.

In this PR:
- add the `theia/keymap` packages that let users change the default keybindings to their preference (see image below)
image

- refactor the settings menu: from a single entry it's now submenu containing both the settings and the keymap actions to open the relevant functionalities (see images below)
image
 
(window version)
image

### CODE REVIEW

**Comment by kittaakos** [comment]
I wonder if the `Settings` name is appropriate here when the window title shows `Preferences`
Screen Shot 2021-03-19 at 15 52 22

and it is Preferences in the Java IDE:

Screen Shot 2021-03-19 at 15 53 22
Screen Shot 2021-03-19 at 15 53 31

What is the current policy? In the past, keeping the Java IDE UI/UX was a rule of thumb? Thanks!

**Review by kittaakos — APPROVED** [review]
It works nicely and the code looks good 👍 

Please wait for another verification before the merge. Thanks!

File: arduino-ide-extension/src/browser/theia/keymaps/keymaps-frontend-contribution.ts
Comment (kittaakos) [reply arduino-ide-extension/src/browser/theia/keymaps/keymaps-frontend-contribution.ts:19]: > There is a related question above 👆

Here: https://github.com/arduino/arduino-ide/pull/231#discussion_r597730519

**Comment by fstasi** [comment]
> It works nicely and the code looks good 👍
> 
> Please wait for another verification before the merge. Thanks!

adding @ubidefeo for a double check before merging

**Review by ubidefeo — COMMENTED** [review]
I have only one problem with this, and it's the placement of the key-bindings as a sub-menu of "preferences".
Historically the IDE preferences menu item served as an item,  not a container of sub-options.
When a new user reads "go to ... > preferences" they won't know wether settings or key-bindings is the place to go.
@giannicolab @per1234 what's your take on this?

**Comment by per1234** [comment]
My concern with the new menu structure is that it breaks all the documentation/tutorials/etc. that tell people to open **File > Preferences**. 

My preference would be to have the keyboard shortcut editor accessed via the **File > Preferences** dialog. But I'm guessing this is simply not technically feasible. So I'm thinking the best approach is to leave the **File > Preferences** menu path alone and add a new menu item for the keyboard shortcuts.

I think it's really nice to have configurable keyboard shortcuts. This is a feature I always look for in the applications I use.

**Comment by fstasi** [comment]
@per1234 @giannicolab @ubidefeo updated as below
image

**Review by per1234 — APPROVED** [review]
Thanks Francesco!

**Pull request merged**
